x-pack/plugin/search-business-rules/src/main/java/org/apache/lucene/search/CappedScorer.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.apache.lucene.search;
+
+import java.io.IOException;
+
+public class CappedScorer extends Scorer {
```

Comment:
maybe extend FilterScorer instead, which already takes care of delegating iterator() or docID()?